### ISSUE TITLE
Node type name clashes with File nodes

### ISSUE THREAD

**Issue by ajayns** [gatsbyjs/gatsby#5182]
While working on a source-plugin which pulls data from a CMS, I've used node type names same as the type names in the CMS. So when I add `gatsby-source-filesystem` to the project for handling remote files and so on, there comes the issue that nodes of type `File` are owned by this plugin while at the same time, the CMS also uses nodes of type `File`. What would be the best approach resolving this issue? Is there any way of easily tweaking `gatsby-source-filesystem`? 

TL;DR Name clash of node of type 'File' when adding `gatsby-source-filesystem` to a CMS source-plugin

**Comment by KyleAMathews** [comment]
Generally when a source provides arbitrary names, to avoid conflicts with other plugins, you should namespace the types e.g. MySourceFile. Changing gatsby-source-filesystem's type name isn't an option.

**Comment by ajayns** [comment]
@KyleAMathews Thanks for the quick response, highly appreciated!
